File: Tema 05 - Typescript/PetShelterFrontend/src/models/Pet.ts
import type { Person } from "./Person";

export class Pet {
    name: string;
    imageUrl: string;
    description: string;
    birthdate: Date;
    type: string;
    rescuer: Person;
    adopter?: Person;
    wieghtInKg: number

    constructor(name: string, imageUrl: string, type: string, description:string, birthdate: Date, wieghtInKg: number, rescuer: Person, adopter?: Person) {
        this.name = name;
        this.imageUrl=imageUrl;
        this.type = type;
        this.description = description;
        this.birthdate = birthdate;
        this.rescuer = rescuer;
        this.adopter = adopter;
        this.wieghtInKg = wieghtInKg;
    }
}

File: Tema 03 - Web API/PetShelter/PetShelter.Domain/Pet.cs
namespace PetShelter.Domain;

public class Pet : PetInfo, INamedEntity
{
    public int Id { get; }

    public PetType Type { get; }

    public Person Rescuer { get; set; }

    public Person Adopter { get; set; }

    public Pet(PetType type, int id = 0)
    {
        Type = type;
        Id = id;
    }
}